### PR TITLE
feature(DSG 2587): Display a list of conversations relating to a transaction in the agency portal

### PR DESCRIPTION
**Pull request by thechechogomez**
# Basics

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

Ticket: https://dol-ewf.atlassian.net/browse/DSG-2587

It allows an agency portal user to view the conversations related to a transaction. 

## Screenshot of the affected changes

https://github.com/Nuvalence/dsgov-web/assets/60438413/8addaaa8-733b-461f-8b6c-7b2148767579

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/agency/feature/dashboard/src/lib/components/transaction-detail/transaction-detail.service.ts [hunk 1]
```diff
@@ -452,5 +455,6 @@ export class TransactionDetailService {
     this._eventsPagination.next(new PagingResponseModel());
     this._notes.next([]);
     this.clearCustomerProvidedDocuments();
+    this._messagingService.cleanUp();
```
Comment (thechechogomez) [inline libs/agency/feature/dashboard/src/lib/components/transaction-detail/transaction-detail.service.ts:458]: I'm not sure if I should clean up here or on the transaction detail component. The component calls this clean up method, so in essence they are the same.

File: libs/shared/data-access/work-api/src/lib/models/conversation/conversation.model.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  displayName: string;
+}
+
+export interface IConversation {
+  id: string;
+  subject: string;
+  createdTimestamp: string;
+  lastUpdatedTimestamp: string;
+  totalMessages: number;
+  unReadMessages: number;
```
Comment (asultan001) [inline libs/shared/data-access/work-api/src/lib/models/conversation/conversation.model.ts:23]: We won't implement read/unread functionality as part of DSG for the time being so you may be able to get rid of this for now.

File: libs/shared/feature/messaging/src/lib/services/messaging.service.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        value: this._referenceId.getValue(),
+      },
+    ];
+
+    return this._workApiRoutesService.getConversations$(requestFilters, pagingRequestModel).pipe(
+      tap(response => {
+        const conversationModels = response.items.map(converstaion => new ConversationModel(converstaion));
+        // TODO: Add display name
+        // conversationModels.forEach(async conversation => {
+        //   conversation.createdByDisplayName = await firstValueFrom(this._userStateService.getUserDisplayName$(conversation.originalMessage.sender));
```
Comment (davidyoung77) [inline libs/shared/feature/messaging/src/lib/services/messaging.service.ts:39]: Can we handle this todo now?

File: libs/agency/feature/dashboard/src/lib/components/transaction-detail/transaction-detail.service.ts (hunk 1 above)
Comment (davidyoung77) [reply libs/agency/feature/dashboard/src/lib/components/transaction-detail/transaction-detail.service.ts:458]: It makes sense here to keep the messaging service a dependency of only this service and not the component

**Review by davidyoung77 — APPROVED** [review]
Nice, approved pending some comments

File: libs/shared/feature/messaging/src/lib/services/messaging.service.ts (hunk 3 above)
Comment (thechechogomez) [reply libs/shared/feature/messaging/src/lib/services/messaging.service.ts:39]: Yeah, that comment should go. The names are brought back from the api

**Review by davidyoung77 — APPROVED** [review]

**Pull request merged**